### ISSUE TITLE
Make revrand algorithm interface compatible with scikit learn

### ISSUE THREAD

**Issue by dsteinberg** [NICTA/revrand#72]
- [x] glm and regression algorithms
- [x] basis functions?

I imagine the basis function parameters will have to be part of the glm/regression objects, and maybe a wrapper can be made so they can be used in the "fit/transform/predict" interface of the scikit learn pipeline tools.

**Comment by dsteinberg** [comment]
Is it worth subclassing `BaseEstimator` from https://github.com/scikit-learn/scikit-learn/blob/master/sklearn/base.py?

**Comment by dsteinberg** [comment]
Have a look at:
https://github.com/NICTA/uncover-ml/blob/master/uncoverml/models.py

**Comment by ltiao** [comment]
Yeah. We should take advantage of [multiple inheritance](https://docs.python.org/2/tutorial/classes.html#multiple-inheritance) making liberal use of the various mixins (`LinearModel`, `RegressorMixin`, `ClassifierMixin`, etc.)

**Comment by dsteinberg** [comment]
I think we just need the `BaseEstimator` class, the others give very specific functionality for the scikit learn linear models (unless I'm overlooking something?). The `BaseEstimator` essentially provides the `get_params()` and `set_params()` functionality -- is that (with our own fit and predict functions) then enough to use the pipeline tools?

**Comment by dsteinberg** [comment]
Actually, the mixins don't seem to give much functionality apart from type/instance testing, so I guess there is not harm in inheriting them
